### ISSUE TITLE
Not needing release mode

### ISSUE THREAD

**Issue by slime73** [love2d/love#584]
**[Original report](https://bitbucket.org/rude/love/issue/579) by hahawoo (Bitbucket: [hahawoo](https://bitbucket.org/hahawoo), GitHub: [hahawoo](https://github.com/hahawoo)).**

----------------------------------------

So I was thinking...

Maybe release mode is a bit... presuming? And adds some unnecessary complexity to configuration?

Release mode seems to assume your game is either in one of two modes; either it's in testing mode, or it's in release mode. I wouldn't be surprised if this often isn't the case though. Perhaps you also want a mode for the demo version, and/or a mode for testers, or something else.

Also, if it wasn't for release mode, the author and url configuration settings wouldn't be necessary, and title wouldn't have two different uses. And the release mode error handler only uses the title, author and url, perhaps the lover might also want to have an email address in the error message, or something different entirely. Surely creating a custom error message with whatever text they want is not too time consuming.

Having the save directory not be in the LOVE subdirectory is restricted to games which are in release mode and are also fused. Perhaps this could simply be a configuration option?

Having separate error handlers for different modes could simply be done by reassigning the name of a single error handler to whatever error handling function the lover chooses:

    :::lua
    love.errhand = my_error_handler_for_testers

Basically I think it assumes too much and doesn't "do more with less".

So what I currently think could be a good idea is...

* Adding a configuration option for using a save directory not in the LOVE subdirectory.
* Removing `release`, `author` and `url` from the configuration options.
* Renaming `title` to `caption` or maybe `screen.caption` to be consistent with `love.graphics.setCaption`.
